webpack-subresource-integrity/src/util.ts

```diff
@@ -241,15 +253,19 @@ export function replaceInSource(
   compiler: Compiler,
   source: sources.Source,
   path: string,
-  replacements: Iterable<[string, string]>
+  replacements: Map<string, string>
 ): sources.Source {
-  const oldSource = source.source();
+  const oldSource = source.source() as string;
   const newAsset = new compiler.webpack.sources.ReplaceSource(source, path);
 
-  for (const [fromString, toString] of replacements) {
-    const pos = oldSource.indexOf(fromString);
-    if (pos >= 0) {
-      newAsset.replace(pos, pos + fromString.length - 1, toString);
+  for (const match of oldSource.matchAll(placeholderRegex)) {
+    const placeholder = match[0];
+    if (placeholder) {
```

Comment:
How about
```suggestion
    const placeholderIndex = match[0]?.index;
    if (placeholderIndex !== undefined) {
```

Then the assertions below are not necessary.